Asterix/tests/test_coronagraph.py

```diff
@@ -15,16 +15,23 @@ def test_default_coronagraph():
     modelconfig = config["modelconfig"]
     Coronaconfig = config["Coronaconfig"]
 
-    # Update the pixels across the pupil
-    modelconfig.update({'diam_pup_in_pix': 80})
-    # Define a round pupil in the apodization plane
-    Coronaconfig.update({'filename_instr_apod': "RoundPup"})
+    # Set coronagraph to be tested
+    coros_to_test = ["fqpm", "wrapped_vortex", "classiclyot", "knife", "hlc"]#, "vortex]  # Vortex is only non-perfect coronagraph left.
+    expected_attenuation = [1e-20, 1e-5, 1e-2, 1e-1, 1e-2]  # Note that these are for the 80 px pupil below
 
-    # Create the coronagraph
-    corono = Coronagraph(modelconfig, Coronaconfig)
-    coro_psf = corono.todetector_intensity(center_on_pixel=True)
+    for i, coro in enumerate(coros_to_test):
+        Coronaconfig.update({"corona_type": coro})
 
-    assert np.max(coro_psf) == 0.0, "A perfect coronagraph should return an empty array."
+        # Update the pixels across the pupil
+        modelconfig.update({'diam_pup_in_pix': 80})
+        # Define a round pupil in the apodization plane
+        Coronaconfig.update({"filename_instr_apod": "RoundPup"})
+
+        # Create the coronagraph
+        corono = Coronagraph(modelconfig, Coronaconfig)
+        coro_psf = corono.todetector_intensity(center_on_pixel=True, in_contrast=True)
+
+        assert np.max(coro_psf) < expected_attenuation[i], f"Attenuation of '{coro}' not below expected {expected_attenuation[i]}."
```

Comment:
you can also test symetry

Reply:
Of the PSF? Indeed, let me add that.

Reply:
... only along one axis for the knife-edge coro though...

Reply:
The knife-edge is complicated as its up-down symmetry is much true to a much higher tolerance than expected, so I am tabling that one for now.